### PR TITLE
Validation model spec

### PR DESCRIPTION
**Pull request by brytebee**
### Validation, test, N+1 solved! 🧪 
In this PR, I have achieved the following;
- Added validations for 
   - User model
   - Post model
- Added Unit Tests
- Added includes() to fix N+1 problem in
   - Model
   - Controllers
   - Views

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
```
Comment (micheaol) [inline spec/models/post_spec.rb:2]: Please ensure that there are unit tests for the `comment, and the `Like` models according to the project requirement.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'Name must not be blank' do
+    subject.name = 'Anny'
+    expect(subject).to be_valid
+  end
+
+  it 'Post counter must be an integer' do
```
Comment (micheaol) [inline spec/models/user_spec.rb:16]: Some of the `tests are failing with the error message below:

image

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @brytebee , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/models/user_spec.rb (hunk 2 above)
Comment (brytebee) [reply spec/models/user_spec.rb:16]: @micheaol Please these are not the tests I wrote for this milestone.

Kindly review the tests for this milestone using the command `rspec spec/models` to review the current milestone.

Thanks for understanding.

Comment (brytebee) [reply spec/models/user_spec.rb:16]: I have updated the README accordingly.

Comment (micheaol) [reply spec/models/user_spec.rb:16]: Hi @brytebee,

I am only highlighting one of the errors received after running your tests.

**Review by micheaol — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hi @brytebee , 

Great work on making the changes requested by the previous reviewer 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 
- You set up linter files and no errors. 💯 

well-1

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**